core/Notification/Manager.php

```diff
@@ -111,10 +114,16 @@ private static function checkId($id)
     private static function addNotification($id, Notification $notification)
     {
         self::saveNotificationAcrossUiRequestsIfNeeded($id, $notification);
+        
+        if (count(self::$notifications) > self::MAX_NOTIFICATIONS_IN_SESSION) {
+            return false;
+        }
```

Comment:
The method `removeOldestNotificationsIfThereAreTooMany` is used before to remove older notifications that exceed the limit. Is this check actually needed? Also it would prevent to add a new notification instead of discarding the oldest.

Reply:
Yes, the method removes the notifications from the session, not from the in memory array. So the current request can still insert 1000 notifications into the session. Then the next request will fail to read the session, long before this code gets invoked again.

Reply:
But wouldn't it make more sense to discard the oldest notification instead of not adding new ones?

Reply:
It could, but I think the added value is minimal there. Just because a notification was displayed earlier in the same request, doesn't mean it no longer needs to be displayed to the user. If it gets to the point that there are too many notifications to show to the user, then the user will simply not see all of them. Most notifications get triggered again anyway, so if some are dealt with, the others should also show up.